spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -85,34 +83,26 @@ public static Object[][] parameters() {
     };
   }
 
-  private final int formatVersion;
+  @Parameter(index = 3)
+  protected int formatVersion;
+
   private final String sourceTableName = "source_table";
   private File fileTableDir;
 
-  public TestAddFilesProcedure(
-      String catalogName, String implementation, Map<String, String> config, int formatVersion) {
-    super(catalogName, implementation, config);
-    this.formatVersion = formatVersion;
-  }
-
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir protected Path temp;
 
-  @Before
+  @BeforeEach
   public void setupTempDirs() {
-    try {
-      fileTableDir = temp.newFolder();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
+    fileTableDir = temp.toFile();
```

Comment:
could you please check if we actually need `fileTableDir`? I think the calls that use `fileTableDir.getAbsolutePath()` can be replaced with `temp.toFile().getAbsolutePath()`.

Reply:
I believe `temp.toFile().getAbsolutePath()` is evaluated in the sql and I couldn't find any errors based on my several attempts. However, `fileTableDir` is called 33 times in this class, and keep the current one is less changes. What do you think?

Reply:
ok let's keep `fileTableDir` in this case